ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSshell.cpp

```diff
@@ -90,21 +91,25 @@ string execute_shell_for_output(const string &command) {
         TranslateMessage(&msg);
         DispatchMessage(&msg);
       }
+      DWORD dwRead = 0;
+      DWORD dwAvail = 0;
+      PeekNamedPipe(hStdOutPipeRead, NULL, 0, NULL, &dwAvail, NULL);
+      if (dwAvail) {
+        char buffer[dwAvail];
```

Comment:
This is a variable-length array, VLAs are not supported in standard C++. You would need to call new here or use a fixed size buffer and read in a loop, as I noted in the below comment.

Reply:
fundies hates C's `new` keyword so I will honor his wishes and use std::vector. That makes it easier, anyway...

Reply:
C does not have a `new` operator, C++ does, C has malloc.

Reply:
Well all i know is he doesn't like C approaches and he doesn't like new so i thought the reason he didn't like new was because it was tied to c. apparently not.

Reply:
The vector is neater because then you don't have to `delete` it.